scripts/us_census/pep/Population_Estimate_by_ASR/county_1990_2000.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''
+This Python Script is
+for County Level Data
+1990-2000.
+'''
+import os
+import numpy as np
+import pandas as pd
+from common_functions import _replace_age
+
+
+def county1990():
+    '''
+    This Python Script Loads csv datasets
+    from 1990-2000 on a County Level,
+    cleans it and create a cleaned csv.
+    '''
+    # Contains the final data which has been taken directly and aggregated.
+    final_df = pd.DataFrame()
+    # Contains aggregated data for age and sex.
+    df_as = pd.DataFrame()
+    # Contains aggregated data for age and race.
+    df_ar = pd.DataFrame()
+    # Reading the csv input file.
+    for i in range(1, 57):
+        if i not in [3, 7, 14, 43, 52]:
+            j = f'{i:02}'
+            url = 'https://www2.census.gov/programs-surveys'+\
+                '/popest/tables/1990-2000/counties/asrh/casrh'+str(j)+'.txt'
+
+            cols=['Year','geo_ID','Race',0,1,2,3,4,5,6,7\
+                ,8,9,10,11,12,13,14,15,16,17]
+            df = pd.read_table(url,index_col=False,delim_whitespace=True\
+                ,skiprows=16,skipfooter=14,engine='python',names=cols,\
+                    encoding='ISO-8859-1')
+            # Removing the lines that have false symbols.
+            num_df = (df.drop(cols, axis=1).join(df[cols]\
+                .apply(pd.to_numeric, errors='coerce')))
+            df = num_df[num_df[cols].notnull().all(axis=1)]
+            df[1:] = df[1:].apply(pd.to_numeric)
+            df['geo_ID'] = df['geo_ID'].astype(int)
+
+            # Providing geoId to the dataframe.
+            df['geo_ID'] = [f'{x:05}' for x in df['geo_ID']]
+            # Replacing the numbers with more understandable metadata headings.
+            df = _replace_age(df)
+
+            # Columns after 11 where having origin.
```

Comment:
Could everything from Line 61-84, for example, be in their own helpful function (to help de-clutter this one function)? They are all replace() function calls.

Reply:
I have changed this to a single replace command using a dictionary.
df = df.replace({'Race':Dictionary values})